crates/storage/src/schema/revision_0055.rs

```diff
@@ -0,0 +1,26 @@
+use anyhow::Context;
+
+pub(crate) fn migrate(tx: &rusqlite::Transaction<'_>) -> anyhow::Result<()> {
+    tx.execute_batch(
+        r"
+        CREATE TABLE trie_class_removals (
+            idx INTEGER PRIMARY KEY,
+            block_number INTEGER
+        );
```

Comment:
I think this becomes a lot simpler and smaller (less database indices required) if we just store the indices as a vector (blob). We should only need all of the indices for a single block.

```suggestion
        CREATE TABLE trie_class_removals (
            block_number INTEGER PRIMARY KEY,
            indices BLOB,
        );
```

Reply:
This would also let you put them all in a single table since they can all share the same block number row.

Unclear if this really matters. It will if they ever set `k` to a large enough number 🤷

Reply:
> Why do you want to put them all in the same table? This would make table updates slightly complex (e.g. when class tries are removed, look up a value for that block, if it's missing create a new one, then update it with the removed indices and store it back, vs. simply adding a new row to a table right now).

In my head this would all be a single operation: insert all node indices for block `i`, remove all indices for block `i-k`. But sure it doesn't have to be one table, not really saving much I don't think.

Reply:
Each block, every one of those tries gets updated. So you could have a single function taking in all three, and performing both the insert and the deletion. I think currently this is done separately via the three different transaction `xxx_trie` calls - these could just be a single one if we wanted.

But I also don't think having three tables tracking these is a problem - we can also merge them later if its an issue. Only "cost" is 3x primary key (block number + db index), which should be much lower than the blobs we are storing alongside.

So you're welcome to leave it as is - I think it was just the layout I had in my header originally to have less tables flying about.